work16/src/main.js

```diff
@@ -0,0 +1,251 @@
+import './style.scss';
+import { differenceInCalendarDays } from 'date-fns';
+
+const REQUEST_URL = "https://mocki.io/v1/71fc7cba-21ec-4437-a8a4-65a3eaa49321";
+
+// 間違っているURLの場合↓
+// const REQUEST_URL = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// 空の配列の場合↓
+// const REQUEST_URL = "https://mocki.io/v1/242b685f-a3d7-45a8-aeca-0376bd495b89";
+// 503エラーの場合↓
+// const REQUEST_URL = "https://httpstat.us/503";
+
+const init = async () => {
+  renderLoadingImg();
+  let responseJsonData;
+
+  try {
+    responseJsonData = await request();
+
+    if (!responseJsonData) {
+      return;
+    }
+  } catch (error) {
+    displayErrorMessage(error.message);
+  }
+  finally {
+    hideLoadingImg();
+  }
+
+  if (!responseJsonData.data.length) {
+    displayErrorMessage("data is empty");
+    return;
+  }
+  renderNewsTab(responseJsonData.data);
+  renderNewsContent(responseJsonData.data);
+  clickedTabEvent(responseJsonData.data);
+}
+
+const request = async () => {
+  const response = await fetch(REQUEST_URL, {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+  if (!response.ok) {
+    displayErrorMessage(`${response.status}:${response.statusText}`);
+  } else {
+    return response.json();
+  }
+}
+
+const displayErrorMessage = (error) => {
+  const p = document.createElement("p");
+  p.textContent = error;
+  document.body.prepend(p);
+}
+
+const renderLoadingImg = () => {
+  const div = document.createElement("div");
+  const img = document.createElement("img");
+
+  div.classList = "loading_wrap";
+  div.id = "js_loading_wrap"
+  img.src = "https://i.postimg.cc/k2RLrq7B/loading-circle.gif";
+  img.alt = "ローディング画像"
+
+  div.appendChild(img);
+  document.body.prepend(div);
+}
+
+const renderNewsTab = (newsDataArray) => {
+  const tabList = document.getElementById("js_tabList");
+  const fragment = document.createDocumentFragment();
+
+  newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    let isInitialDisplay = newsCategoryObj.isInitialDisplay;
+
+    li.ariaSelected = isInitialDisplay;
+    li.setAttribute("roll", "tab");
+    li.id = `${"js_tabTopics" + ++newsCategoryIndex}`;
+    li.classList = "tabTopics";
+
+    a.textContent = newsCategoryObj.field
+    a.href = "#"
+
+    li.appendChild(a);
+    fragment.appendChild(li);
+  });
+  tabList.appendChild(fragment);
+}
+
+const renderNewsContent = (newsDataArray) => {
+  newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+    const tabTopics = document.querySelectorAll(".tabTopics");
+    const newsSection = createSection(tabTopics, newsCategoryIndex);
+    const newsContentsData = newsCategoryObj.contents;
+    const newsCategoryImg = newsCategoryObj.img;
+    let isInitialDisplay = newsCategoryObj.isInitialDisplay;
+
+    if (isInitialDisplay === "true") {
```

Comment:
Q.This differs from the previous commitments.
https://github.com/yasu-pro/js-handson/pull/18/commits/e58e83b07136394ca81dda8e3c990b38a2a5e36a

Reply:
Sorry.
- My mistake about commit [d6e8417]( https://github.com/yasu-pro/js-handson/commit/d6e8417e730dddcb6e2b0db83733b997ff9c4e52).

- For ` if (isInitialDisplay === "true") { `, the value of isInitialDisplay in work16.json was of type string. So I changed the value of isInitialDisplay to a boolean type and then changed the if statement to ` if (isInitialDisplay) { `.

commit: [7e8b1a8](https://github.com/yasu-pro/js-handson/pull/18/commits/7e8b1a85919b74e8df1acd8344027b7fcec08e01)